### PR TITLE
[For 2.6.3] Feature/2687  improve email quality

### PR DESCRIPTION
**Pull request by inkhey**
related to #2687 

Many mail optimization:
- Refactoring: Unique code for creating Message(EmailNotificationMessage)
- Add somes headers for best spam checker notation: Message-ID, Date
- Add "Content-Language" header for semantic
- Add "X-Auto-Response-Suppress" and "Auto-Submitted" headers in order to avoid receiving autoreply message.
- Add automatically generated text version of the html email with **html2text** (markdown like mail) and tracim **html sanitizer** (clean html used for conversion) for spam checker notation.

## Checkpoints

The goal of this section is to help code reviewer and merge request author to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewer**

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

**For developper**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] Original issue have been updated with a short resume of implemented solution

**For quality team**

- [x] Tested by quality team

### CODE REVIEW

File: backend/tracim_backend/tests/functional/test_mail_notification.py [hunk 1]
```diff
@@ -114,7 +164,7 @@ def test_func__create_new_content_with_notification__ok__nominal_case(
         uapi = user_api_factory.get(current_user=None)
         current_user = uapi.get_one_by_email("admin@admin.admin")
         # set admin as french, useful to verify if i18n work properly
-        current_user.lang = "fr"
+        current_user.lang = "en"
```
Comment (PhilippeAccorsi) [inline backend/tracim_backend/tests/functional/test_mail_notification.py:167]: Comment explain to set in french but parameter is set to english.

Comment (inkhey) [reply backend/tracim_backend/tests/functional/test_mail_notification.py]: you're right, that's a mistake.

**Comment by PhilippeAccorsi** [comment]
**Tested and verified on branch**. Result here https://www.mail-tester.com/test-bykdfa8ed
Note 9/10 , and green result for spamassassin 

Without this fix, we have note 5.6/10 and result here https://www.mail-tester.com/test-8wadroii5 (-3.4 for spamassassin).

File: backend/tracim_backend/lib/mail_notifier/notifier.py [hunk 2]
```diff
@@ -148,7 +147,7 @@ def __init__(self, smtp_config: SmtpConfiguration, config: CFG, session: Session
         #     session_factory = get_session_factory(engine)
         #     app_config = CFG(settings)
 
-    def _get_sender(self, user: User = None) -> str:
+    def _get_sender(self, user: User = None) -> EmailAddress:
         """
         Return sender string like "Bob Dylan
```
Comment (raphj) [inline backend/tracim_backend/lib/mail_notifier/notifier.py:152]: This docstring probably needs to be updated

File: backend/tracim_backend/lib/mail_notifier/utils.py [hunk 3]
```diff
@@ -24,3 +37,65 @@ class EST(object):
...
+    def from_rfc_email_address(cls, rfc_email: str) -> "EmailAddress":
+        label, email = parseaddr(rfc_email)
+        return EmailAddress(label, email)
+
+    @property
+    def address(self):
+        formatted_address = formataddr((self.label, self.email))
+        if not self.force_angle_bracket or self.label:
+            return formatted_address
+        else:
```
Comment (raphj) [inline backend/tracim_backend/lib/mail_notifier/utils.py:58]: You may get rid of this "else:" if you wish

Comment (inkhey) [reply backend/tracim_backend/lib/mail_notifier/utils.py]: done

File: backend/tracim_backend/lib/mail_notifier/notifier.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/lib/mail_notifier/notifier.py]: done

**Pull request merged**